### DISCUSSION TITLE
Add ATOMIC attribute

### DISCUSSION THREAD

**Discussion by HenryHRich** [jsoftware/jsource#251] [Ideas]
Have ATOMIC@ATOMIC use @: code

If a verb is ATOMIC, reblock very large arguments for best use of caches

**Comment by moon-chilled** [comment]
does this need a new attribute?  should just need u rank<=v rank

> reblock very large arguments for best use of caches

ideal solution to this is a compiler, to fuse everything.  Obvious difficulties

**Comment by HenryHRich** [comment]
It needs a minimum-rank attribute for a verb, to make sure the system
doesn't subdivide a unit.

**Comment by HenryHRich** [comment]
atomic@atomic now gives a warning; breaking rank loops is an ongoing project
